### PR TITLE
Resident Evil 4 Mobile Edition

### PR DESCRIPTION
**Pull request by alborrajo**
# Tested versions

- 1.00.00

# Known issues

- Reloading by using the accelerometer doesn't work
- Doesn't play video cutscenes

# Screenshots

imagen
imagen

### CODE REVIEW

**Comment by omagadsamsa** [comment]
When I press the settings button the game crashes

**Comment by omagadsamsa** [comment]
in this build there are no sound effects in many games

**Comment by omagadsamsa** [comment]
image
There is still sound stuttering in your build

**Comment by omagadsamsa** [comment]
in the previous build the sound effects returned

**Comment by omagadsamsa** [comment]
omg
image

**Comment by bmwm111** [comment]
hi please help mi how to download games touchhle emulator instaled my phone , game download please

**Comment by bmwm111** [comment]
Screenshot_20240225_092511

**Comment by omagadsamsa** [comment]
build for android no work re4

File: src/frameworks/av_audio/av_audio_player.rs [hunk 1]
```diff
@@ -81,6 +80,9 @@ pub const CLASSES: ClassExports = objc_classes! {
 
     let host_object = env.objc.borrow_mut::<AVAudioPlayerHostObject>(this);
     host_object.audio_file_url = url;
+
+    // TODO: Check for errors, return nil and write them to error if there are
```
Comment (hikari-no-yume) [inline src/frameworks/av_audio/av_audio_player.rs:84]: test of per-commit github reviews

Comment (hikari-no-yume) [reply src/frameworks/av_audio/av_audio_player.rs:84]: *puts on testing hat* i think you should change this

**Review by hikari-no-yume — CHANGES_REQUESTED** [review]
test

File: src/frameworks/av_audio/av_audio_player.rs (hunk 1 above)
Comment (hikari-no-yume) [reply src/frameworks/av_audio/av_audio_player.rs:84]: Cool, so I can actually leave reviews on just a single commit in a pull and the comment _will_ show up on the pull request and _won't_ forget the context?

**Comment by thundergun1** [comment]
I've noticed a couple of issues on Android. Firstly, whenever a use prompt appears (Open Door, Raise Ladder etc) a big chunk of geometry disappears, though reappears when the prompt leaves.

Also, the accelerometer doesn't seem to work, as even on the highest shake sensitivity I can't reload by shaking my phone.

Apologies for possibly nagging on both of your PRs, it's fantastic work overall I just thought I'd point out some issues I've had

Screenshot_20240321_081024_touchHLE
Screenshot_20240321_080810_touchHLE

**Comment by alborrajo** [comment]
I've been playing on Android and I haven't seen this issue before. Considering you also had the other issue with the RE Degeneration splash screen which i couldn't replicate, couldn't be an issue on your Android version? Please provide more details about the device you're using.

About reloading, i have my suspicion that is related to the issue #151 as I can't reload on my phone either.

**Comment by thundergun1** [comment]
Of course. I'm using a Samsung Galaxy A42 with an Adreno GPU running Android 12.

I'm unsure regarding the accelerometer. I currently have 3 different apks of TouchHLE saved (one for your RE4 PR, one for your Degeneration PR, and one for the trunk) and on the latter 2 the accelerometer works perfectly on any game that uses it, whereas on the apk for the RE4 PR it doesn't, even for games like CNK3D where it works great on the other builds

**Review by totvrey5 — CHANGES_REQUESTED** [review]
Ok

**Comment by thundergun1** [comment]
Leon seems to have stopped existing as of this newest build, though his gun is still present.

Screenshot_20240407_115657_touchHLE

Also it may be that the disappearing graphics I mentioned earlier could be an issue with Adreno GPUs rather than an issue with your PR

**Comment by alborrajo** [comment]
> Leon seems to have stopped existing as of this newest build, though his gun is still present.

Try again with the latest build

**Comment by thundergun1** [comment]
Can confirm Leon is back

**Comment by DrPonx** [comment]
could u comebine both resident evil 4 and Degeneration into one workflow

**Comment by thundergun1** [comment]
Although the accelerometer bug was fixed as of a few commits ago, it still doesn't seem to work on the latest build of your PR

**Comment by celerizer** [comment]
Here is a slightly playable controller mapping for anyone interested (use UI layout 1):
```
jp.co.capcom.res4: --landscape-left --button-to-touch=DPadUp,72,198 --button-to-touch=DPadDown,72,292 --button-to-touch=DPadLeft,24,246 --button-to-touch=DPadRight,120,246 --button-to-touch=X,440,210 --button-to-touch=LeftShoulder,405,280 --button-to-touch=A,370,210 --button-to-touch=B,440,140 --button-to-touch=Y,370,140
```

**Comment by alborrajo** [comment]
This branch is now merged with trunk 🎉

**Pull request merged**